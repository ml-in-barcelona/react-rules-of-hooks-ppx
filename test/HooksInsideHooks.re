module GQL = {
  let useMutation =
      (~redirectSuccess=None, ~redirectCancel=None, ~onboarding=false, ()) => {
    let (mutation, _, _) = ApolloHooks.useMutation(~variables, definition);

    () => {
      mutation()
      ->Utils.Promise.map(fst)
      ->Utils.Apollo.Mutation.trackResult(
          ~analyticsMessage=
            onboarding ? "Clicked Pro Plan" : "Clicked Manage Billing",
        )
      ->ignore;
    };
  };
};
