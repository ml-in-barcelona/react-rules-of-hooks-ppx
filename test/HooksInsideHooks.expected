module GQL =
  struct
    let useMutation ?(redirectSuccess= None)  ?(redirectCancel= None)
      ?(onboarding= false)  () =
      let (mutation,_,_) = ApolloHooks.useMutation ~variables definition in
      fun ()  ->
        (((((mutation ()) |. Utils.Promise.map) fst) |.
            Utils.Apollo.Mutation.trackResult)
           ~analyticsMessage:(match onboarding with
                              | true  ->
                                  (("Clicked Pro Plan")[@reason.raw_literal
                                                         "Clicked Pro Plan"])
                              | false  ->
                                  (("Clicked Manage Billing")[@reason.raw_literal
                                                               "Clicked Manage Billing"]))
           ())
          |. ignore
  end